tests/mock_virtual_machine.h

```diff
@@ -37,6 +37,11 @@ struct MockVirtualMachine : public multipass::VirtualMachine
         ON_CALL(*this, management_ipv4()).WillByDefault(Return("0.0.0.0"));
         ON_CALL(*this, get_all_ipv4(_)).WillByDefault(Return(std::vector<std::string>{"192.168.2.123"}));
         ON_CALL(*this, ipv6()).WillByDefault(Return("::/0"));
+        ON_CALL(*this, ensure_vm_is_running()).WillByDefault(Invoke([]() { throw std::runtime_error("Not running"); }));
```

Comment:
Did you forget this one? Or is it needed?

Reply:
I added it because it is what the stub does. If I remove it, the list tests fail.

I investigated why it is needed. It happens that the function `ensure_vm_is_running()` is called many times when executing things in the instance. In particular, when we use `send_commands()` the test client hangs up waiting for something from the VM, which in fact is a mock and will return from an execution on the terminal.

I added an expectation in the body of the test function, so as to not add things to the mock.